update/Pos/Update/Poll/Trans.hs

```diff
@@ -111,9 +110,8 @@ instance MonadPollRead m =>
             map (\dps -> (hash $ upsProposal $ dpsUndecided dps, dps)) <$>
             getDeepProposals cd
     getBlockIssuerStake e = lift . getBlockIssuerStake e
-    getSlottingData = ether $ do
-        new <- pmSlottingData <$> get
-        maybe getSlottingData pure new
+    getEpochSlottingData ei = ether $
+        MM.lookupM getEpochSlottingData ei =<< use pmSlottingDataL
```

Comment:
You forgot to redefine `getEpochLastIndex`. It should take `pmSlottingDataL` into account as well.